dialogue-apache-hc4-client/src/main/metrics/dialogue-apache-client-metrics.yml

```diff
@@ -0,0 +1,11 @@
+options:
+  javaPackage: com.palantir.dialogue.hc4
+  javaVisibility: packagePrivate
+namespaces:
+  dialogue.client.pool:
```

Comment:
The metric name doesn't reference the apache client, I tried to keep it as general as possible so we could implement it for other clients in the future if applicable. Thoughts?

Reply:
I like the idea of having consistent metric regardless of underlying client implementation. We can update the docs if we decide to work on another client